forge-game/src/main/java/forge/game/GameEntityView.java

```diff
@@ -45,35 +46,34 @@ protected void updateName(GameEntity e) {
     public int getPreventNextDamage() {
         return get(TrackableProperty.PreventNextDamage);
     }
-    protected void updatePreventNextDamage(GameEntity e) {
+    public void updatePreventNextDamage(GameEntity e) {
         set(TrackableProperty.PreventNextDamage, e.getPreventNextDamageTotalShields());
     }
 
     public Iterable<CardView> getAttachedCards() {
-        return get(TrackableProperty.AttachedCards);
+        if (hasAnyCardAttachments()) {
+            Iterable<CardView> active = Iterables.filter(get(TrackableProperty.AttachedCards), c -> !c.isPhasedOut());
+            if (!Iterables.isEmpty(active)) {
+                return active;
+            }
+        }
+        return null;
```

Comment:
does this causes the "unattach" event be on the different timing?

Reply:
I will check but for this I need to fix a funny bug first:
![grafik](https://github.com/Card-Forge/forge/assets/8506892/08a7349c-e06e-49ae-b1eb-cd16f7951658)

First the middle aura will phase out indirectly but because it was also granted phasing it will then phase back in
now it is unattached and both will move to the GY

Reply:
Well I could not observe any change in behaviour
So I'd conclude the GUI updates for phasedOut CardView property happen in the right order